Add interface to display newspaper issues missing a dateIssued

whikloj opened the pull request:
Hopefully addresses some of [ISLANDORA-1181](https://jira.duraspace.org/browse/ISLANDORA-1181)

whikloj committed:
Add interface to display newspaper issues missing a dateIssued

whikloj committed:
Coder fixes

ppound commented on includes/manage_newspaper.inc:
```diff
@@ -19,10 +19,139 @@ require_once dirname(__FILE__) . '/utilities.inc';
+      $issue_obj = islandora_object_load($issue['pid']);
+      if ($issue_obj) {
+        $escaped_pid = str_replace(':', '_', $issue['pid']);
+        $escaped_pid = $issue['pid'];
```
this looks odd should, probably don't need line 51 and 52 probably just one or the other?

whikloj replied:
Thank you, that was a hold over. I'll remove it.

ppound commented on includes/utilities.inc:
```diff
@@ -232,3 +232,60 @@ function islandora_newspaper_get_date_issued(AbstractObject $object) {
+  if (is_null($object)) {
+    $object = islandora_get_tuque_connection();
+  }
+  $results = $object->repository->ri->sparqlQuery($query);
```
Should we check for xacml and pid namespace etc.?

whikloj replied:
Were you thinking of some sort of special checks or would implementing [these same filters](https://github.com/Islandora/islandora_solution_pack_newspaper/blob/7.x/includes/utilities.inc#L99-L114) be sufficient?

ppound replied:
I think the same filters would do.

whikloj committed:
Call hooks to limit to newspapers viewable to the user.
Allow submitting some dates instead of requiring all at once.
Display form with "No issues found" instead of blank.

whikloj committed:
Updated UI text.

willtp87 commented on css/islandora_newspaper.base.css:
```diff
@@ -0,0 +1,8 @@
+}
+.islandora-newspaper-empty-issues .form-item.form-type-textfield input {
+    margin-right: 10px;
+}
```
All files should have a new line here for git.

willtp87 commented on includes/manage_newspaper.inc:
```diff
@@ -19,7 +19,148 @@ require_once dirname(__FILE__) . '/utilities.inc';
+        $form['dates_table'][$escaped_pid] = array(
+          '#type' => 'textfield',
+          '#name' => "issue_dates[$escaped_pid]",
+          '#prefix' => t('<div class="islandora-newspaper-empty-issues">'),
```
I don't think this sort of thing should be exposed for translation.

willtp87 commented on includes/manage_newspaper.inc:
```diff
@@ -19,7 +19,148 @@ require_once dirname(__FILE__) . '/utilities.inc';
+          $dt = new DateTime($date);
+          islandora_newspaper_set_date_issued($object, $dt);
+        }
+        catch (Exception $e) {
```
Generally squashing all exceptions is inadvisable.  If there is a good reason for it then it should be documented to prevent someone from changing it.

willtp87 commented on includes/utilities.inc:
```diff
@@ -232,3 +232,77 @@ function islandora_newspaper_get_date_issued(AbstractObject $object) {
+EOQ;
+
+  if (is_null($object)) {
+    $object = islandora_get_tuque_connection();
```
Variable re-use; going from a Fedora object to a Tuque connection.

willtp87 commented on includes/utilities.inc:
```diff
@@ -232,3 +232,77 @@ function islandora_newspaper_get_date_issued(AbstractObject $object) {
+EOQ;
+  }
+  $query = <<<EOQ
+PREFIX islandora-rels-ext: <http://islandora.ca/ontology/relsext#>
```
Constants at https://github.com/Islandora/tuque/blob/1.x/FedoraRelationships.php#L9-L12

whikloj committed:
Changes from code review.

ppound commented:
I've commented on the jira ticket here https://jira.duraspace.org/browse/ISLANDORA-1181

daniel-dgi commented:
See https://jira.duraspace.org/browse/ISLANDORA-1181 for resolution from committer's call.

whikloj committed:
Add dateIssued to the MODS data stream if it exists.

whikloj committed:
Coder fixes

whikloj committed:
Default form not seeing the new dateIssued because it has an @encoding attribute.

whikloj committed:
Fix deletes and finding appropriate insertion point

ppound commented on includes/utilities.inc:
```diff
@@ -232,3 +232,137 @@ function islandora_newspaper_get_date_issued(AbstractObject $object) {
+  // hook as it will return some things we don't want. Perhaps change the return
+  // structure in the future to specify which module they are coming from? For
+  // now we will just get XACML's directly.
+  $query_optionals = (array) module_invoke('islandora_xacml_api', 'islandora_basic_collection_get_query_optionals', 'view');
```
i should have noticed this sooner but I think we should change view to manage in the call to islandora_basic_collection_get_query_optionals.  We want the user to be able to set the date so they will need manage access.

whikloj committed:
Make sure we have the correct query for XACML permissions.

ppound commented:
Pulled this down and tested and it works as expected.

whikloj committed:
Various code review fixes

whikloj committed:
Not needed as I just want the contents of the array.
